File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamChiTiet/SanPhamChiTietColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamChiTiet")]
[BasedOnRow(typeof(SanPhamChiTietRow), CheckNames = true)]
public class SanPhamChiTietColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    [EditLink]
    public string TrongLuongText { get; set; }
    public int SoLanDungMin { get; set; }
    public int SoLanDungMax { get; set; }
    public string HamLuongProtein { get; set; }
    public string Calories { get; set; }
    public string HamLuongBcaa { get; set; }
    public string XuatXu { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamChiTietColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamChiTietRow } from "./SanPhamChiTietRow";

export interface SanPhamChiTietColumns {
    Id: Column<SanPhamChiTietRow>;
    SanPhamTenSp: Column<SanPhamChiTietRow>;
    TrongLuongText: Column<SanPhamChiTietRow>;
    SoLanDungMin: Column<SanPhamChiTietRow>;
    SoLanDungMax: Column<SanPhamChiTietRow>;
    HamLuongProtein: Column<SanPhamChiTietRow>;
    Calories: Column<SanPhamChiTietRow>;
    HamLuongBcaa: Column<SanPhamChiTietRow>;
    XuatXu: Column<SanPhamChiTietRow>;
}

export class SanPhamChiTietColumns extends ColumnsBase<SanPhamChiTietRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamChiTiet';
    static readonly Fields = fieldsProxy<SanPhamChiTietColumns>();
}